src/hotspot/share/nmt/regionsTree.hpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+#ifndef NMT_REGIONSTREE_HPP
+#define NMT_REGIONSTREE_HPP
+
+#include "logging/log.hpp"
+#include "nmt/nmtCommon.hpp"
+#include "nmt/vmatree.hpp"
+#include "nmt/virtualMemoryTracker.hpp"
+
+// RegionsTree extends VMATree to add some more specific API and also defines a helper
+// for processing the tree nodes in a shorter and more meaningful way.
+class RegionsTree : public VMATree {
+ private:
+  NativeCallStackStorage _ncs_storage;
+  bool _with_storage;
+
+ public:
+  RegionsTree(bool with_storage) : VMATree() , _ncs_storage(with_storage), _with_storage(with_storage) { }
+
+  ReservedMemoryRegion find_reserved_region(address addr);
+
+  SummaryDiff commit_region(address addr, size_t size, const NativeCallStack& stack);
+  SummaryDiff uncommit_region(address addr, size_t size);
+
+  using Node = VMATree::TreapNode;
+
+  class NodeHelper {
```

Comment:
Most of the methods here should be `const` and take `const NodeHelper&` as arguments.

Reply:
Done.